azure-pipelines.yml

```diff
@@ -0,0 +1,90 @@
+parameters:
+- name: module_names
+  type: object
+  default:
+  - nidigital
+  - nitclk  
+  - nifgen
+  - nidcpower
+  - nidmm
+  - niscope
+  - nimodinst
+  - nise
+  - niswitch
+
+# A pipeline with no CI trigger
+trigger: none
+
+# specific path
+pr:
+  branches:
+    include:
+    - master
+    - releases/*
+  paths:
+    exclude:
+    - CHANGELOG.md
+    - CONTRIBUTING.md
+    - .gitattributes
+    - LICENSE
+    - VERSION
+  drafts: false
+
+stages:
+
+  - stage: ensure_code_gen_up_to_date
+    pool:
+      name: nimi-bot
+      demands:
+        - nimi-bot -equals True
+        - agent.os -equals Linux
+    jobs:
+    - job: ensure_code_gen_up_to_date
+      workspace:
+        clean: all
+      steps:
+        - checkout: self  
+          clean: true
+          fetchDepth: $(clone_depth)
+        - task: PythonScript@0
+          displayName: Ensure codegen up to date
```

Comment:
Job is `ensure_code_gen_up_to_date`, but here you type "codegen".
Let's be precise and correct:

* Abbreviation is "codegen" as used throughout the repo. So no underscore between code and gen.
* Either say codegen here, or spell out "code generation"
* Up to date should be "up-to-date"
* Is this self-explanatory enough? displayName could be better: "Ensure changes to code generation are committed to repository".

Reply:
> * Up to date should be "up-to-date": we can't use '-' in job name. 

> * Is this self-explanatory enough? displayName could be better: "Ensure changes to code generation are committed to repository".
I have updated the display name as per your suggestion.
Previously, I was using naming conventions from other PR's e.g https://ni.visualstudio.com/DevCentral/_build/results?buildId=1352531&view=logs&j=abca7640-95f0-558f-0764-dca33dd54edb&t=f9f9b2ab-d6da-505d-8ca9-db34651b59c7